src/transport/HttpOutboundTransporter.ts

```diff
@@ -33,31 +33,48 @@ export class HttpOutboundTransporter implements OutboundTransporter {
   }
 
   public async sendMessage(outboundPackage: OutboundPackage) {
-    const { payload, endpoint, responseRequested } = outboundPackage
+    const { connection, payload, endpoint } = outboundPackage
 
     if (!endpoint) {
       throw new Error(`Missing endpoint. I don't know how and where to send the message.`)
+      return
+    }
+    // TODO: use mediation config for queue logic
+    if (endpoint == 'didcomm:transport/queue') {
+      this.logger.debug('Storing message for queue: ', { connection, payload })
+      connection.assertReady()
+      if (connection && connection.theirKey) {
+        this.agent.mediator.queueMessage(connection.theirKey, payload)
+      }
+      return
     }
-
     this.logger.debug(
       `Sending outbound message to connection ${outboundPackage.connection.id}`,
       outboundPackage.payload
     )
-
     try {
-      const response = await fetch(endpoint, {
+      const response = fetch(endpoint, {
         method: 'POST',
         body: JSON.stringify(payload),
         headers: { 'Content-Type': this.agentConfig.didCommMimeType },
       })
-      const responseMessage = await response.text()
+
+      const action = await Promise.race([
+        response,
+        new Promise((accept) => {
+          setTimeout(() => accept(false), 15000)
+        }),
+      ])
+      const responseMessage = action ? await (action as Response).text() : false
```

Comment:
`Promise.race` will return the first promise that executes first, I assume it will clean up all other promises accordingly. My assumption is that the first issue of timeout not being cleared is that the timeout is cleared by the promise. The second issue is also the same case for promise.race cleaning up. I assumed if the timer returned first, the promise.race would clear the fetch. I do not know if my assumptions are correct. I will try my hand at AbortController.

I also found this link that's interesting and related. https://stackoverflow.com/a/46376473

Reply:
We could also use rxjs: https://rxjs.dev/api/fetch/fromFetch

We can them match it with a timeout operator that automatically cancels the request

Reply:
It doesn't clean up other promises. I was also curious about that if it's the behavior of race. Couldn't find any specific mention in MDN docs, just this article [Use Promise.race to Timeout Promises](https://javascript.plainenglish.io/use-promise-race-to-timeout-promises-6710cb0a3164), part Caveat: Cleanup. So I tested myself :) 

https://codepen.io/jakubkoci/pen/WNjxgQR?editors=1011

Reply:
this fetch has been updated to use an `abortController`. I believe that addresses the two issues at hand.